File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function postorderTraversal(root: TreeNode | null): number[] {
    let res: number[] = []

    const postorder = (node: TreeNode | null): void => {
        if (node === null) return

        postorder(node.left)
        postorder(node.right)
        res.push(node.val)
    }

    postorder(root)
    return res
};

File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        res = []
        
        def dfs(node: TreeNode) -> None:
            if not node:
                return 

            if node.left:
                dfs(node.left)

            if node.right:
                dfs(node.right)

            res.append(node.val)

        dfs(root)

        return res